### ISSUE TITLE
Correct  Sublist of Elements of Index / Correct List of Fields in SubIndex of and Index  is not returened 

### ISSUE THREAD

**Issue by abuech** [spyder-ide/spyder#19755]
### What steps will reproduce the problem?

datensatz=["Tom",26,1.84]
print (datensatz[1:2]

Result:
Instead of field 1 + 2  only field one is displayed 

print (datensatz[1:2])
[26]

Array of lists: 

datensatz1 = ["Tom",26,1.84]
datensatz2 = ["Lisa",27,1.92]
datensatz5 = [[datensatz1],[datensatz2]]
print(datensatz5[0][:1])

Result:
Instead of "26" (Field 1 of dataset 0) the whole dataset is displayed

print(datensatz5[0][:1])
[['Tom', 26, 1.84]]

## Versions

* Spyder version: 5.3.3  (conda)
* Python version: 3.9.12 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Darwin 21.1.0

### Dependencies

```
# Mandatory:
applaunchservices >=0.3.0     :  0.3.0 (OK)
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.31.1;<8.0.0       :  7.31.1 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.16.0 (OK)
keyring >=17.0.0              :  23.4.0 (OK)
nbconvert >=4.0               :  6.4.4 (OK)
numpydoc >=0.6.0              :  1.4.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0;<3.0           :  2.14.5 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.5.0;<1.6.0          :  1.5.0 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.3.2;<5.4.0      :  5.3.2 (OK)
qtpy >=2.1.0                  :  2.2.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  63.4.1 (OK)
sphinx >=0.6.6                :  5.0.2 (OK)
spyder_kernels >=2.3.3;<2.4.0 :  2.3.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
zmq >=22.1.0                  :  23.2.0 (OK)

# Optional:
cython >=0.21                 :  0.29.32 (OK)
matplotlib >=3.0.0            :  3.5.2 (OK)
numpy >=1.7                   :  1.21.5 (OK)
pandas >=1.1.1                :  1.4.4 (OK)
scipy >=0.17.0                :  1.7.3 (OK)
sympy >=0.7.3                 :  1.10.1 (OK)
```

### Environment

<details><summary>Environment</summary>

```
# packages in environment at /Users/abuech/opt/anaconda3:
#
# Name                    Version                   Build  Channel
_anaconda_depends         2022.05                  py39_0  
_ipyw_jlab_nb_ext_conf    0.1.0            py39hecd8cb5_1  
absl-py                   1.2.0                    pypi_0    pypi
aiohttp                   3.8.1            py39hca72f7f_1  
aiosignal                 1.2.0              pyhd3eb1b0_0  
alabaster                 0.7.12             pyhd3eb1b0_0  
anaconda                  custom                   py39_1  
anaconda-client           1.11.0           py39hecd8cb5_0  
anaconda-navigator        2.3.0            py39hecd8cb5_0  
anaconda-project          0.11.1           py39hecd8cb5_0  
anyio                     3.5.0            py39hecd8cb5_0  
appdirs                   1.4.4              pyhd3eb1b0_0  
applaunchservices         0.3.0            py39hecd8cb5_0  
appnope                   0.1.2           py39hecd8cb5_1001  
appscript                 1.1.2            py39h9ed2024_0  
argon2-cffi               21.3.0             pyhd3eb1b0_0  
argon2-cffi-bindings      21.2.0           py39hca72f7f_0  
arrow                     1.2.2              pyhd3eb1b0_0  
astroid                   2.11.7           py39hecd8cb5_0  
astropy                   5.1              py39h67323c0_0  
asttokens                 2.0.5              pyhd3eb1b0_0  
astunparse                1.6.3                    pypi_0    pypi
async-timeout             4.0.2            py39hecd8cb5_0  
atomicwrites              1.4.0                      py_0  
attrs                     21.4.0             pyhd3eb1b0_0  
automat                   20.2.0                     py_0  
autopep8                  1.6.0              pyhd3eb1b0_1  
babel                     2.9.1              pyhd3eb1b0_0  
backcall                  0.2.0              pyhd3eb1b0_0  
backports                 1.1                pyhd3eb1b0_0  
backports.functools_lru_cache 1.6.4              pyhd3eb1b0_0  
backports.tempfile        1.0                pyhd3eb1b0_1  
backports.weakref         1.0.post1                  py_1  
bcrypt                    3.2.0            py39hca72f7f_1  
beautifulsoup4            4.11.1           py39hecd8cb5_0  
binaryornot               0.4.4              pyhd3eb1b0_1  
bitarray                  2.5.1            py39hca72f7f_0  
bkcharts                  0.2              py39hecd8cb5_1  
black                     22.6.0           py39hecd8cb5_0  
blas                      1.0                         mkl  
bleach                    4.1.0              pyhd3eb1b0_0  
blosc                     1.21.0               h8346a28_1  
bokeh                     2.4.3            py39hecd8cb5_0  
boto3                     1.24.28          py39hecd8cb5_0  
botocore                  1.27.28          py39hecd8cb5_0  
bottleneck                1.3.5            py39h67323c0_0  
brotli                    1.0.9                hca72f7f_7  
brotli-bin                1.0.9                hca72f7f_7  
brotlipy                  0.7.0           py39h9ed2024_1003  
brunsli                   0.1                  h23ab428_0  
bzip2                     1.0.8                h1de35cc_0  
c-ares                    1.18.1               hca72f7f_0  
ca-certificates           2022.07.19           hecd8cb5_0  
cachetools                4.2.2              pyhd3eb1b0_0  
certifi                   2022.9.14        py39hecd8cb5_0  
cffi                      1.15.1           py39hc55c11b_0  
cfitsio                   3.470                hee0f690_6  
chardet                   4.0.0           py39hecd8cb5_1003  
charls                    2.2.0                h23ab428_0  
charset-normalizer        2.0.4              pyhd3eb1b0_0  
click                     8.0.4            py39hecd8cb5_0  
cloudpickle               2.0.0              pyhd3eb1b0_0  
clyent                    1.2.2            py39hecd8cb5_1  
colorama                  0.4.5            py39hecd8cb5_0  
colorcet                  3.0.0            py39hecd8cb5_0  
conda                     22.9.0           py39hecd8cb5_0  
conda-build               3.21.8           py39hecd8cb5_2  
conda-content-trust       0.1.3            py39hecd8cb5_0  
conda-env                 2.6.0                         1  
conda-pack                0.6.0              pyhd3eb1b0_0  
conda-package-handling    1.9.0            py39hca72f7f_0  
conda-repo-cli            1.0.20           py39hecd8cb5_0  
conda-token               0.4.0              pyhd3eb1b0_0  
conda-verify              3.4.2                      py_1  
constantly                15.1.0             pyh2b92418_0  
cookiecutter              1.7.3              pyhd3eb1b0_0  
cryptography              37.0.1           py39hf6deb26_0  
cssselect                 1.1.0              pyhd3eb1b0_0  
curl                      7.84.0               hca72f7f_0  
cycler                    0.11.0             pyhd3eb1b0_0  
cython                    0.29.32          py39he9d5cce_0  
cytoolz                   0.11.0           py39h9ed2024_0  
daal4py                   2021.6.0         py39h01d92e1_1  
dal                       2021.6.0           haf03e11_929  
dask                      2022.2.1           pyhd3eb1b0_0  
dask-core                 2022.2.1           pyhd3eb1b0_0  
dataclasses               0.8                pyh6d0b6a4_7  
datashader                0.14.1           py39hecd8cb5_0  
datashape                 0.5.4            py39hecd8cb5_1  
debugpy                   1.5.1            py39he9d5cce_0  
decorator                 5.1.1              pyhd3eb1b0_0  
defusedxml                0.7.1              pyhd3eb1b0_0  
diff-match-patch          20200713           pyhd3eb1b0_0  
dill                      0.3.4              pyhd3eb1b0_0  
distributed               2022.2.1           pyhd3eb1b0_0  
docutils                  0.18.1           py39hecd8cb5_3  
entrypoints               0.4              py39hecd8cb5_0  
et_xmlfile                1.1.0            py39hecd8cb5_0  
executing                 0.8.3              pyhd3eb1b0_0  
filelock                  3.6.0              pyhd3eb1b0_0  
flake8                    4.0.1              pyhd3eb1b0_1  
flask                     1.1.2              pyhd3eb1b0_0  
flatbuffers               22.9.24                  pypi_0    pypi
fonttools                 4.25.0             pyhd3eb1b0_0  
freetype                  2.11.0               hd8bbffd_0  
frozenlist                1.2.0            py39hca72f7f_0  
fsspec                    2022.7.1         py39hecd8cb5_0  
future                    0.18.2           py39hecd8cb5_1  
gast                      0.4.0                    pypi_0    pypi
gensim                    4.1.2            py39he9d5cce_0  
gettext                   0.21.0               h7535e17_0  
giflib                    5.2.1                haf1e3a3_0  
glib                      2.69.1               h8346a28_1  
glob2                     0.7                pyhd3eb1b0_0  
gmp                       6.2.1                he9d5cce_3  
gmpy2                     2.1.2            py39hd5de756_0  
google-api-core           2.10.1           py39hecd8cb5_0  
google-auth               2.6.0              pyhd3eb1b0_0  
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-cloud-core         2.2.2              pyhd3eb1b0_0  
google-cloud-storage      1.43.0           py39hecd8cb5_0  
google-crc32c             1.1.2            py39h9ed2024_0  
google-pasta              0.2.0                    pypi_0    pypi
google-resumable-media    1.3.1              pyhd3eb1b0_1  
googleapis-common-protos  1.56.4           py39hecd8cb5_0  
greenlet                  1.1.1            py39h23ab428_0  
grpcio                    1.42.0           py39ha29bfda_0  
gst-plugins-base          1.14.0               h4180768_2  
gstreamer                 1.14.0               h0fc69c2_2  
h5py                      3.7.0            py39h4a1dd59_0  
hdf5                      1.10.6               hdbbcd12_0  
heapdict                  1.0.1              pyhd3eb1b0_0  
holoviews                 1.15.0           py39hecd8cb5_0  
hvplot                    0.8.0            py39hecd8cb5_0  
hyperlink                 21.0.0             pyhd3eb1b0_0  
icu                       58.2                 h0a44026_3  
idna                      3.3                pyhd3eb1b0_0  
imagecodecs               2021.8.26        py39h0f85e6e_1  
imageio                   2.19.3           py39hecd8cb5_0  
imagesize                 1.4.1            py39hecd8cb5_0  
importlib-metadata        4.11.3           py39hecd8cb5_0  
importlib_metadata        4.11.3               hd3eb1b0_0  
incremental               21.3.0             pyhd3eb1b0_0  
inflection                0.5.1            py39hecd8cb5_0  
iniconfig                 1.1.1              pyhd3eb1b0_0  
intake                    0.6.5              pyhd3eb1b0_0  
intel-openmp              2021.4.0          hecd8cb5_3538  
intervaltree              3.1.0              pyhd3eb1b0_0  
ipykernel                 6.15.2           py39hecd8cb5_0  
ipython                   7.31.1           py39hecd8cb5_1  
ipython_genutils          0.2.0              pyhd3eb1b0_1  
ipywidgets                7.6.5              pyhd3eb1b0_1  
isort                     5.9.3              pyhd3eb1b0_0  
itemadapter               0.3.0              pyhd3eb1b0_0  
itemloaders               1.0.4              pyhd3eb1b0_1  
itsdangerous              2.0.1              pyhd3eb1b0_0  
jdcal                     1.4.1              pyhd3eb1b0_0  
jedi                      0.18.1           py39hecd8cb5_1  
jellyfish                 0.9.0            py39hca72f7f_0  
jinja2                    2.11.3             pyhd3eb1b0_0  
jinja2-time               0.2.0              pyhd3eb1b0_3  
jmespath                  0.10.0             pyhd3eb1b0_0  
joblib                    1.1.0              pyhd3eb1b0_0  
jpeg                      9e                   hca72f7f_0  
jq                        1.6               h9ed2024_1000  
json5                     0.9.6              pyhd3eb1b0_0  
jsonschema                4.16.0           py39hecd8cb5_0  
jupyter                   1.0.0            py39hecd8cb5_8  
jupyter_client            7.3.5            py39hecd8cb5_0  
jupyter_console           6.4.3              pyhd3eb1b0_0  
jupyter_core              4.11.1           py39hecd8cb5_0  
jupyter_server            1.18.1           py39hecd8cb5_0  
jupyterlab                3.4.4            py39hecd8cb5_0  
jupyterlab_pygments       0.1.2                      py_0  
jupyterlab_server         2.10.3             pyhd3eb1b0_1  
jupyterlab_widgets        1.0.0              pyhd3eb1b0_1  
jxrlib                    1.1                  haf1e3a3_2  
keras                     2.10.0                   pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
keyring                   23.4.0           py39hecd8cb5_0  
kiwisolver                1.4.2            py39he9d5cce_0  
krb5                      1.19.2               hcd88c3b_0  
lazy-object-proxy         1.6.0            py39h9ed2024_0  
lcms2                     2.12                 hf1fd2bf_0  
lerc                      3.0                  he9d5cce_0  
libaec                    1.0.4                hb1e8313_1  
libarchive                3.6.1                he336d3b_0  
libbrotlicommon           1.0.9                hca72f7f_7  
libbrotlidec              1.0.9                hca72f7f_7  
libbrotlienc              1.0.9                hca72f7f_7  
libclang                  14.0.6                   pypi_0    pypi
libcrc32c                 1.1.1                hb1e8313_2  
libcurl                   7.84.0               h6dfd666_0  
libcxx                    14.0.6               h9765a3e_0  
libdeflate                1.8                  h9ed2024_5  
libedit                   3.1.20210910         hca72f7f_0  
libev                     4.33                 h9ed2024_1  
libffi                    3.3                  hb1e8313_2  
libgfortran               3.0.1                h93005f0_2  
libiconv                  1.16                 hca72f7f_2  
libidn2                   2.3.2                h9ed2024_0  
liblief                   0.11.5               he9d5cce_1  
libllvm11                 11.1.0               h46f1229_5  
libllvm12                 12.0.0               h9b2ccf5_3  
libnghttp2                1.46.0               ha29bfda_0  
libpng                    1.6.37               ha441bb4_0  
libpq                     12.9                 h1c9f633_3  
libprotobuf               3.20.1               h8346a28_0  
libsodium                 1.0.18               h1de35cc_0  
libspatialindex           1.9.3                h23ab428_0  
libssh2                   1.10.0               h0a4fc7d_0  
libtiff                   4.4.0                h2ef1027_0  
libunistring              0.9.10               h9ed2024_0  
libwebp                   1.2.2                h56c3ce4_0  
libwebp-base              1.2.2                hca72f7f_0  
libxml2                   2.9.14               hbf8cd5e_0  
libxslt                   1.1.35               h5b33f42_0  
libzopfli                 1.0.3                hb1e8313_0  
llvm-openmp               14.0.6               h0dcd299_0  
llvmlite                  0.38.0           py39h8346a28_0  
locket                    1.0.0            py39hecd8cb5_0  
lxml                      4.9.1            py39h65b224f_0  
lz4-c                     1.9.3                h23ab428_1  
lzo                       2.10                 haf1e3a3_2  
markdown                  3.3.4            py39hecd8cb5_0  
markupsafe                2.0.1            py39h9ed2024_0  
matplotlib                3.5.2            py39hecd8cb5_0  
matplotlib-base           3.5.2            py39hfb0c5b7_0  
matplotlib-inline         0.1.6            py39hecd8cb5_0  
mccabe                    0.7.0              pyhd3eb1b0_0  
mistune                   0.8.4           py39h9ed2024_1000  
mkl                       2021.4.0           hecd8cb5_637  
mkl-service               2.4.0            py39h9ed2024_0  
mkl_fft                   1.3.1            py39h4ab4a9b_0  
mkl_random                1.2.2            py39hb2f4e1b_0  
mock                      4.0.3              pyhd3eb1b0_0  
mpc                       1.1.0                h6ef4df4_1  
mpfr                      4.0.2                h9066e36_1  
mpi                       1.0                       mpich  
mpich                     3.3.2                hc856adb_0  
mpmath                    1.2.1            py39hecd8cb5_0  
msgpack-python            1.0.3            py39haf03e11_0  
multidict                 5.2.0            py39hca72f7f_3  
multipledispatch          0.6.0            py39hecd8cb5_0  
munkres                   1.1.4                      py_0  
mypy_extensions           0.4.3            py39hecd8cb5_1  
navigator-updater         0.2.1                    py39_1  
nbclassic                 0.3.5              pyhd3eb1b0_0  
nbclient                  0.5.13           py39hecd8cb5_0  
nbconvert                 6.4.4            py39hecd8cb5_0  
nbformat                  5.5.0            py39hecd8cb5_0  
ncurses                   6.3                  hca72f7f_3  
nest-asyncio              1.5.5            py39hecd8cb5_0  
networkx                  2.8.4            py39hecd8cb5_0  
nltk                      3.7                pyhd3eb1b0_0  
nose                      1.3.7           pyhd3eb1b0_1008  
notebook                  6.4.12           py39hecd8cb5_0  
nspr                      4.33                 he9d5cce_0  
nss                       3.74                 h47edf6a_0  
numba                     0.55.1           py39hae1ba45_0  
numexpr                   2.8.3            py39h2e5f0a9_0  
numpy                     1.21.5           py39h2e5f0a9_3  
numpy-base                1.21.5           py39h3b1a694_3  
numpydoc                  1.4.0            py39hecd8cb5_0  
oauthlib                  3.2.1                    pypi_0    pypi
olefile                   0.46               pyhd3eb1b0_0  
oniguruma                 6.9.7.1              h9ed2024_0  
openjpeg                  2.4.0                h66ea3da_0  
openpyxl                  3.0.10           py39hca72f7f_0  
openssl                   1.1.1q               hca72f7f_0  
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 21.3               pyhd3eb1b0_0  
pandas                    1.4.4            py39he9d5cce_0  
pandocfilters             1.5.0              pyhd3eb1b0_0  
panel                     0.13.1           py39hecd8cb5_0  
param                     1.12.0             pyhd3eb1b0_0  
parsel                    1.6.0            py39hecd8cb5_0  
parso                     0.8.3              pyhd3eb1b0_0  
partd                     1.2.0              pyhd3eb1b0_1  
pathlib                   1.0.1              pyhd3eb1b0_1  
pathspec                  0.9.0            py39hecd8cb5_0  
patsy                     0.5.2            py39hecd8cb5_1  
pcre                      8.45                 h23ab428_0  
pep8                      1.7.1            py39hecd8cb5_1  
pexpect                   4.8.0              pyhd3eb1b0_3  
pickleshare               0.7.5           pyhd3eb1b0_1003  
pillow                    9.2.0            py39hde71d04_1  
pip                       22.2.2           py39hecd8cb5_0  
pkginfo                   1.8.2              pyhd3eb1b0_0  
platformdirs              2.5.2            py39hecd8cb5_0  
plotly                    5.9.0            py39hecd8cb5_0  
pluggy                    1.0.0            py39hecd8cb5_1  
ply                       3.11             py39hecd8cb5_0  
poyo                      0.5.0              pyhd3eb1b0_0  
prometheus_client         0.14.1           py39hecd8cb5_0  
prompt-toolkit            3.0.20             pyhd3eb1b0_0  
prompt_toolkit            3.0.20               hd3eb1b0_0  
protego                   0.1.16                     py_0  
protobuf                  3.20.1           py39he9d5cce_0  
psutil                    5.9.0            py39hca72f7f_0  
ptyprocess                0.7.0              pyhd3eb1b0_2  
pure_eval                 0.2.2              pyhd3eb1b0_0  
py                        1.11.0             pyhd3eb1b0_0  
py-lief                   0.11.5           py39he9d5cce_1  
pyasn1                    0.4.8              pyhd3eb1b0_0  
pyasn1-modules            0.2.8                      py_0  
pycodestyle               2.8.0              pyhd3eb1b0_0  
pycosat                   0.6.3            py39h9ed2024_0  
pycparser                 2.21               pyhd3eb1b0_0  
pyct                      0.4.8            py39hecd8cb5_1  
pycurl                    7.45.1           py39h0a4fc7d_0  
pydispatcher              2.0.5            py39hecd8cb5_2  
pydocstyle                6.1.1              pyhd3eb1b0_0  
pyerfa                    2.0.0            py39h9ed2024_0  
pyflakes                  2.4.0              pyhd3eb1b0_0  
pygments                  2.11.2             pyhd3eb1b0_0  
pyhamcrest                2.0.2              pyhd3eb1b0_2  
pyjwt                     2.4.0            py39hecd8cb5_0  
pylint                    2.14.5           py39hecd8cb5_0  
pyls-spyder               0.4.0              pyhd3eb1b0_0  
pyobjc-core               8.5              py39hc55c11b_0  
pyobjc-framework-cocoa    8.5              py39hca72f7f_0  
pyobjc-framework-coreservices 8.5              py39hca72f7f_0  
pyobjc-framework-fsevents 8.5              py39hecd8cb5_0  
pyodbc                    4.0.34           py39he9d5cce_0  
pyopenssl                 22.0.0             pyhd3eb1b0_0  
pyparsing                 3.0.9            py39hecd8cb5_0  
pyqt                      5.15.7           py39he9d5cce_0  
pyqt5-sip                 12.11.0          py39he9d5cce_0  
pyqtwebengine             5.15.7           py39he9d5cce_0  
pyrsistent                0.18.0           py39hca72f7f_0  
pysocks                   1.7.1            py39hecd8cb5_0  
pytables                  3.6.1            py39h648f197_1  
pytest                    7.1.2            py39hecd8cb5_0  
python                    3.9.12               hdfd78df_0  
python-dateutil           2.8.2              pyhd3eb1b0_0  
python-fastjsonschema     2.16.2           py39hecd8cb5_0  
python-libarchive-c       2.9                pyhd3eb1b0_1  
python-lsp-black          1.2.1            py39hecd8cb5_0  
python-lsp-jsonrpc        1.0.0              pyhd3eb1b0_0  
python-lsp-server         1.5.0            py39hecd8cb5_0  
python-slugify            5.0.2              pyhd3eb1b0_0  
python-snappy             0.6.0            py39h23ab428_3  
python.app                3                py39hca72f7f_0  
pytz                      2022.1           py39hecd8cb5_0  
pyviz_comms               2.0.2              pyhd3eb1b0_0  
pywavelets                1.3.0            py39hca72f7f_0  
pyyaml                    6.0              py39hca72f7f_1  
pyzmq                     23.2.0           py39he9d5cce_0  
qdarkstyle                3.0.2              pyhd3eb1b0_0  
qstylizer                 0.1.10             pyhd3eb1b0_0  
qt                        5.15.9               hecd8cb5_0  
qt-main                   5.15.2               h719ae48_7  
qt-webengine              5.15.9               h90a370e_4  
qtawesome                 1.0.3              pyhd3eb1b0_0  
qtconsole                 5.3.2            py39hecd8cb5_0  
qtpy                      2.2.0            py39hecd8cb5_0  
qtwebkit                  5.212                h24dc246_4  
queuelib                  1.5.0            py39hecd8cb5_0  
readline                  8.1.2                hca72f7f_1  
regex                     2022.7.9         py39hca72f7f_0  
requests                  2.28.1           py39hecd8cb5_0  
requests-file             1.5.1              pyhd3eb1b0_0  
requests-oauthlib         1.3.1                    pypi_0    pypi
ripgrep                   13.0.0               hc2228c6_0  
rope                      0.22.0             pyhd3eb1b0_0  
rsa                       4.7.2              pyhd3eb1b0_1  
rtree                     0.9.7            py39hecd8cb5_1  
ruamel_yaml               0.15.100         py39h9ed2024_0  
s3transfer                0.6.0            py39hecd8cb5_0  
scikit-image              0.19.2           py39hae1ba45_0  
scikit-learn              1.0.2            py39hae1ba45_1  
scikit-learn-intelex      2021.6.0         py39hecd8cb5_0  
scipy                     1.7.3            py39h8c7af03_0  
scrapy                    2.6.2            py39hecd8cb5_0  
seaborn                   0.11.2             pyhd3eb1b0_0  
send2trash                1.8.0              pyhd3eb1b0_1  
service_identity          18.1.0             pyhd3eb1b0_1  
setuptools                63.4.1           py39hecd8cb5_0  
sip                       6.6.2            py39he9d5cce_0  
six                       1.16.0             pyhd3eb1b0_1  
smart_open                5.2.1            py39hecd8cb5_0  
snappy                    1.1.9                he9d5cce_0  
sniffio                   1.2.0            py39hecd8cb5_1  
snowballstemmer           2.2.0              pyhd3eb1b0_0  
sortedcollections         2.1.0              pyhd3eb1b0_0  
sortedcontainers          2.4.0              pyhd3eb1b0_0  
soupsieve                 2.3.1              pyhd3eb1b0_0  
sphinx                    5.0.2            py39hecd8cb5_0  
sphinxcontrib-applehelp   1.0.2              pyhd3eb1b0_0  
sphinxcontrib-devhelp     1.0.2              pyhd3eb1b0_0  
sphinxcontrib-htmlhelp    2.0.0              pyhd3eb1b0_0  
sphinxcontrib-jsmath      1.0.1              pyhd3eb1b0_0  
sphinxcontrib-qthelp      1.0.3              pyhd3eb1b0_0  
sphinxcontrib-serializinghtml 1.1.5              pyhd3eb1b0_0  
spyder                    5.3.3            py39hecd8cb5_0  
spyder-kernels            2.3.3            py39hecd8cb5_0  
sqlalchemy                1.4.39           py39hca72f7f_0  
sqlite                    3.39.3               h707629a_0  
stack_data                0.2.0              pyhd3eb1b0_0  
statsmodels               0.13.2           py39hca72f7f_0  
sympy                     1.10.1           py39hecd8cb5_0  
tabulate                  0.8.10           py39hecd8cb5_0  
tbb                       2021.6.0             haf03e11_0  
tbb4py                    2021.6.0         py39haf03e11_0  
tblib                     1.7.0              pyhd3eb1b0_0  
tenacity                  8.0.1            py39hecd8cb5_1  
tensorboard               2.10.1                   pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
tensorflow                2.10.0                   pypi_0    pypi
tensorflow-estimator      2.10.0                   pypi_0    pypi
tensorflow-io-gcs-filesystem 0.27.0                   pypi_0    pypi
termcolor                 2.0.1                    pypi_0    pypi
terminado                 0.13.1           py39hecd8cb5_0  
testpath                  0.6.0            py39hecd8cb5_0  
text-unidecode            1.3                pyhd3eb1b0_0  
textdistance              4.2.1              pyhd3eb1b0_0  
threadpoolctl             2.2.0              pyh0d69192_0  
three-merge               0.1.1              pyhd3eb1b0_0  
tifffile                  2021.7.2           pyhd3eb1b0_2  
tinycss                   0.4             pyhd3eb1b0_1002  
tk                        8.6.12               h5d9f67b_0  
tldextract                3.2.0              pyhd3eb1b0_0  
toml                      0.10.2             pyhd3eb1b0_0  
tomli                     2.0.1            py39hecd8cb5_0  
tomlkit                   0.11.1           py39hecd8cb5_0  
toolz                     0.11.2             pyhd3eb1b0_0  
tornado                   6.2              py39hca72f7f_0  
tqdm                      4.64.1           py39hecd8cb5_0  
traitlets                 5.1.1              pyhd3eb1b0_0  
twisted                   22.2.0           py39hca72f7f_1  
typed-ast                 1.4.3            py39h9ed2024_1  
typing-extensions         4.3.0            py39hecd8cb5_0  
typing_extensions         4.3.0            py39hecd8cb5_0  
tzdata                    2022c                h04d1e81_0  
ujson                     5.4.0            py39he9d5cce_0  
unidecode                 1.2.0              pyhd3eb1b0_0  
unixodbc                  2.3.11               hb456775_0  
urllib3                   1.26.11          py39hecd8cb5_0  
w3lib                     1.21.0             pyhd3eb1b0_0  
watchdog                  2.1.6            py39h999c104_0  
wcwidth                   0.2.5              pyhd3eb1b0_0  
webencodings              0.5.1            py39hecd8cb5_1  
websocket-client          0.58.0           py39hecd8cb5_4  
werkzeug                  2.0.3              pyhd3eb1b0_0  
wget                      1.21.3               h6dfd666_0  
whatthepatch              1.0.2            py39hecd8cb5_0  
wheel                     0.37.1             pyhd3eb1b0_0  
widgetsnbextension        3.5.2            py39hecd8cb5_0  
wrapt                     1.14.1           py39hca72f7f_0  
wurlitzer                 3.0.2            py39hecd8cb5_0  
xarray                    0.20.1             pyhd3eb1b0_1  
xlrd                      2.0.1              pyhd3eb1b0_0  
xlsxwriter                3.0.3              pyhd3eb1b0_0  
xlwings                   0.27.15          py39hecd8cb5_0  
xz                        5.2.6                hca72f7f_0  
yaml                      0.2.5                haf1e3a3_0  
yapf                      0.31.0             pyhd3eb1b0_0  
yarl                      1.8.1            py39hca72f7f_0  
zeromq                    4.3.4                h23ab428_0  
zfp                       0.5.5                he9d5cce_6  
zict                      2.1.0            py39hecd8cb5_0  
zipp                      3.8.0            py39hecd8cb5_0  
zlib                      1.2.12               h4dc903c_3  
zope                      1.0              py39hecd8cb5_1  
zope.interface            5.4.0            py39h9ed2024_0  
zstd                      1.5.2                hcb37349_0  

```
</details>

**Comment by ccordoba12** [comment]
Hey @abuech, thanks for reporting. You said

> Instead of field 1 + 2 only field one is displayed

This happens because slicing intervals in Python are semi-open, i.e. they include the first element in the interval but not the last one but the previous element to that.

In your example

> print (datensatz[1:2])

the previous element to `2` is `1`, so that's why you only got `26` as the result. If you want to get up until element `2`, you need to run

> print (datensatz[1:3])

The same is true for the code

> datensatz1 = ["Tom",26,1.84]
> datensatz2 = ["Lisa",27,1.92]
> datensatz5 = [[datensatz1],[datensatz2]]
> print(datensatz5[0][:1])

So, if you want to take up to `26` you need to change your last line to be

> print(datensatz5[0][:2])

**Comment by ccordoba12** [comment]
If you need more help with slicing, please read this tutorial about it:

https://bas.codes/posts/python-slicing

**Comment by ccordoba12** [comment]
Finally, I'm going to close this issue because it's not related to Spyder directly. Please note that this is a place to report internal problems with Spyder, so if you need more help to learn how to program in Python, we recommend you to open your questions in StackOverflow:

https://stackoverflow.com/

**Comment by abuech** [comment]
Hello Carlos

Thank you very much for your support. Pls excuse that I have posted this request into the Spyder formus instead of using a Python3 blog.  I was miss leaded because the kernel was crashing some how (console didn‘t respond anymore) and the message to report a bug popped up. It‘s clear it‘s not your business to provide developer support.

Thank you again.
Best
Alex


büch & partner AG
Alexander Büch
Blauenstrasse 6
4054 Basel
+41 79 320 65 87

Mob: +41 79 320 65 87
________________________________
Von: Carlos Cordoba ***@***.***>
Gesendet: Monday, October 10, 2022 6:00:58 PM
An: spyder-ide/spyder ***@***.***>
Cc: Alexander Büch ***@***.***>; Mention ***@***.***>
Betreff: Re: [spyder-ide/spyder] Correct Sublist of Elements of Index / Correct List of Fields in SubIndex of and Index is not returened (Issue #19755)


Closed #19755<https://github.com/spyder-ide/spyder/issues/19755> as completed.

**Comment by ccordoba12** [comment]
No worries, that's fine as long as in the future it's clear for you what's the right place to ask for help or report bugs.
